docs/test_csv_data_sec_cat_if.yaml

```diff
@@ -5,4 +5,33 @@ attrs:
   sec_cats:
   - Class (class)
   - Type (type)
+time_format: '%Y'
+dimensions:
```

Comment:
Okay, I was not aware that it could be a goal that others publish in the interchange format. I consider that unlikely anyway (our attrs are kind of specific), but human-readability for datasets with homogeneous dimensions is a worthy goal in its own right.

Making the dimensions optional is possible, but we then loop back to the problem with non-explicit date columns versus dimension columns. This is somewhat encoded in the time_format, but we discussed that it would be better to make it explicit. Maybe a good option would be to make the `dimensions` key mandatory, but it can be dict of `entity: dimensions` mappings, or it can just be a list with the dimensions for all entities? That way datasets with homogeneous dimensions don't have to repeat information. A generalization of this idea would be to have a dict but with a catch-all entry like `*` or so.

Examples would be:

**current (homogeneous)**:
```yaml
dimensions:
  CO2:
     - area (ISO3)
     - time
     - category (IPCC2006)
  N2O:
     - area (ISO3)
     - time
     - category (IPCC2006)
```

**current (mixed)**:
```yaml
dimensions:
  CO2:
     - area (ISO3)
     - time
     - category (IPCC2006)
  N2O:
     - area (ISO3)
     - time
     - category (IPCC2006)
  population:
     - area (ISO3)
     - time
```

**with list-only option (homogeneous)**:
```yaml
dimensions:
   - area (ISO3)
   - time
   - category (IPCC2006)
```
with a mixed dataset it would look like the current

**with default key (homogeneous)**:
```yaml
dimensions:
  '*':
     - area (ISO3)
     - time
     - category (IPCC2006)
```

**with default key (mixed)**:
```yaml
dimensions:
  '*':
     - area (ISO3)
     - time
     - category (IPCC2006)
  population:
     - area (ISO3)
     - time
```

I personally don't care much, the list-only option is optimized for datasets with homogeneous dimensions, and as soon as an entity with a different set of dimensions is added, it explodes. The default key option is less readable ("what does the asterisk mean?"), but is pretty neat for the (common?) case of a bazillion different gases and then one or two other entities. I think we're writing the yaml file for computers anyway (the CSV is for humans), so I would slightly prefer the default key option.

Reply:
I think the option with a default and exceptions is the most elegant one. But it's also the hardest to code a writing function for. The option with either a dict (mixed) or a list (homogeneous) is also good.

Reply:
I implemented the default option. Our reading functions are even using it already, as well as `ds.pr.to_interchange_format`. (-: